### PR TITLE
Add disable jit flag to logmatcher

### LINKED ISSUE #2986: Getting Issue of SE Linux while ceating parsers for Syslog-NG
Hi,

I have created parsers for Syslog NG and below are the snippets written in the syslog NG configuration file:

filter sh_auth_logs
{
       program("sshd");
}

Parser p_ssh_auth_logs
{
       db_parser(file("/etc/syslog-ng/patterndb.d/sshd_events.xml"));
}

source syslog_udp
{
       syslog(transport(udp) port(514));
};

template t_ssh { template("$DATE $HOST $MSGHDR$MSG\n"); template_escape(no); };

destination df_wrt0
{
        file("/var/log/syslog-ng.log"
        template(t_ssh)
       owner("root")
        group("root")
        perm(0777));

);

log
{
       source(syslog_udp);
       filter(sh_auth_logs);
       destination(df_wrt0);
       flags(final);
};

 after doing all the above changes and restarting syslog-ng, I am getting the attached error of SE Linux.

Requesting all to provide solution for the same issue.
Syslog NG Error

### PR DESCRIPTION
**Pull request by bazsi**
logmatcher: allow the user to disable PCRE JIT using the disable-jit flag

this should resolve #2986 at least the normal filter() case. the same in multi-line(regexp) support is not this trivial.

**EDIT**:
config examples
```
rewrite r_demo {
  subst("a+" "b" value(".extra.info") flags("disable-jit"));
};

filter f_demo {
  host("internal-.*" flags("disable-jit"));
};

```

### CODE REVIEW

File: lib/logmatcher.c [hunk 1]
```diff
@@ -347,7 +347,9 @@ log_matcher_pcre_re_compile(LogMatcher *s, const gchar *re, GError **error)
     }
 
   /* optimize regexp */
-  self->extra = pcre_study(self->pattern, PCRE_STUDY_JIT_COMPILE, &errptr);
+  self->extra = pcre_study(self->pattern,
```
Comment (Kokan) [inline lib/logmatcher.c]: could you please extract that condition (` (self->super.flags & LMF_DISABLE_JIT) ? 0 : PCRE_STUDY_JIT_COMPILE`) out of the function call and assign to a local `gint options` variable instead ? also `LMF_DISABLE_JIT` should only negate the proper bit in the options and not set to `0`.

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/logmatcher.c (hunk 1 above)
Comment (bazsi) [reply lib/logmatcher.c]: I extracted some functions in the area, can you check if this is ok?

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by bazsi** [comment]
I already have patches in that direction.

**Comment by gaborznagy** [comment]
> what I am working on is a global use-jit() option that would control the
disable-jit flag of filters as well as the jit mode for multi-line mode

Thanks, I wanted to have some thoughts about how should we solve the JIT configuration in other modules before we merge something on a drivel level, which we should support to provide backward compatibility.

**Pull request merged**